lib/erb/compiler.rb

```diff
@@ -331,7 +355,7 @@ def compile(s)
     end
     add_put_cmd(out, content) if content.size > 0
     out.close
-    return out.script, *magic_comment
+    return out, *magic_comment
```

Comment:
I don't think it's a good idea to break the compatibility and return something that's not a source from `src`. Please consider addressing your concern while keeping the current interface.

Reply:
It's probably impossible, the only solution is to make a new interface. Do you prefer that?